onadata/apps/logger/models/instance.py

```diff
@@ -794,11 +794,21 @@ def soft_delete_attachments(self, user=None):
 def post_save_submission(sender, instance=None, created=False, **kwargs):
     """Update XForm, Project, JSON field
 
-    - XForm submission coun
+    - XForm submission count & instances_with_geopoints field
     - Project date modified
     - Update the submission JSON field data
     """
     if instance.deleted_at is not None:
+        # update xform if no instance has geoms
+        if (
+            instance.xform.instances.filter(
+                deleted_at__isnull=True, geom=None
+            ).count()
+            < 1
+        ):
+            instance.xform.instances_with_geopoints = False
+            instance.xform.save()
+
```

Comment:
Let's have this happen in `update_xform_submission_count` and let's try to avoid a count on the XForm. If possible let's use the Forms `submission_count` property